app/models/ingredient.rb

```diff
@@ -1,3 +1,9 @@
 class Ingredient < ApplicationRecord
+  has_many :recipe_ingredients
+  has_many :recipes, through: :recipe_ingredients
    validates_presence_of :name, :cost
+
+   def self.cost_total
+    @total = self.where(recipe: recipe_id).sum(:cost)
```

Comment:
Similar to the other class method, I'm not sure this _needs_ to be a class method... Again, looking at the user story: 

```
I see the total cost of all of the ingredients in the recipe.
```

...which to me reads, I have 1 recipe, and I want to see the total cost of all its ingredients. So I start with a `@recipe` object, and I can get all the ingredietns from taht one instance of Recipe via its association: 

```ruby
@recipe.ingredients
# => [ <# Ingredient 0x0.... @name = "...", @cost = "..." >, <# ... > , <# ... > ]
```

So now, I see an array of all the ingredients. There's a quick way to get the `sum` of all those ingredients' cost using an AR method called `.sum`: 

```
@recipe.ingredients.sum(:cost)
```

In essence, we are writing an instance method in the `Recipe` model where it uses its association to `ingredients`. So we coudl move this method to the `Recipe.rb` file: 


```ruby
# recipe.rb
def total_cost_of_ingredients
   ingredients.sum(:cost)
end
```

Reply:
^ Personally I think that way is easier. But, you CAN do it as a class method, but again you're right that you'd have to specify the `recipe_id`. You set up the `recipe` key, but because this is a many-to-many relationship that's associated to `recipes` plural, we can't use a `recipe` key - the terminal will tell us it doesn't know what `recipe` is. 
If you continue with this class method, you'd have to do a manual join, something like: 

```ruby
   Ingredient.joins(:recipes).where("recipe_id = #{recipe_id}").sum(:cost)
# this works because we have to join on associations. 
```

But as we saw above, this is the long-hand way of doing the `@recipe.ingredients.sum(:cost)` in the Recipe model as an instance method.